boutdata/squashoutput.py

```diff
@@ -215,3 +258,180 @@ def squashoutput(
+            self.parallel = False
+            self.open_files(filenames)
+            return
+
+        if self.parallel is True or self.parallel == 0:
+            from boututils.run_wrapper import determineNumberOfCPUs
+
+            self.parallel = determineNumberOfCPUs()
+
+        self.n_outputs = len(filenames)
+
+        self.create_workers(filenames)
+
+    def __del__(self):
+        if self.parallel is False:
+            for f in self.files:
+                f.close()
+                del f
+        else:
+            for worker, connection, _ in self.workers:
+                # Send None to terminate worker process cleanly
+                connection.send(None)
+                worker.join()
+                connection.close()
+
+    def open_files(self, filenames):
+        """
+        Open files for serial writing
+
+        Parameters
+        ----------
+        filenames : list of str
+            Names of the files to write to.
+        """
+        from boututils.datafile import DataFile
+
+        self.files = [
+            DataFile(name, create=True, write=True, **self.kwargs) for name in filenames
+        ]
+
+    def create_workers(self, filenames):
+        """
+        Create workers for parallel writing
+
+        Parameters
+        ----------
+        filenames : list of str
+            Names of the files to write to.
+        """
+        from multiprocessing import Process, Pipe
+
+        n = min(self.parallel, self.n_outputs)
+        files_per_proc = [self.n_outputs // n for _ in range(n)]
+        for i in range(self.n_outputs % n):
+            files_per_proc[i] += 1
+        assert sum(files_per_proc) == self.n_outputs
+        counter = 0
+        self.workers = []
+        for i in range(n):
+            files_dict = {}
+            worker_indexes = range(counter, counter + files_per_proc[i])
+            for index in worker_indexes:
+                files_dict[index] = filenames[counter]
+                counter = counter + 1
+            parent_connection, child_connection = Pipe()
+            worker = Process(
+                target=self.worker_function,
+                args=(child_connection, files_dict),
+            )
+            worker.start()
+            self.workers.append((worker, parent_connection, worker_indexes))
+
+    def worker_function(self, connection, files_dict):
+        """
+        Function controlling execution on worker processes
+        """
+        from boututils.datafile import DataFile
+        from boututils.boutarray import BoutArray
+
+        output_files = {}
+        for i, name in files_dict.items():
+            output_files[i] = DataFile(name, create=True, write=True, **self.kwargs)
+
+        while True:
+            args = connection.recv()
+            if args is None:
+                # Terminate process cleanly
+                for f in files_dict.values():
+                    f.close()
+                connection.close()
+                return 0
```

Comment:
Is the function expected to return an integer?

Reply:
Doesn't look like it. I think I just assumed before that it would be.
```suggestion
                return
```